src/main.rs

```diff
@@ -1361,6 +1396,25 @@ mod cli {
             <Self as CommandFactory>::command()
         }
     }
+
+    /// The given query arg could be a literal tree-sitter query or a path to a file containing
+    /// a tree-sitter query.
+    #[derive(Debug, Clone)]
+    pub struct CodeQuery(String);
+
+    impl Deref for CodeQuery {
```

Comment:
Fixed in https://github.com/alexpovel/srgn/pull/144/commits/a9e1a7771b44b791c161da31d98dc7eea4aeb5e1

I'm not overly worried about the `Deref` anti-pattern myself but in this case it can easily be replaced with an `impl Into`.